### PR TITLE
Executed excercise 0.5

### PR DESCRIPTION
**Pull request by ArpitAgrawal001**
Here is a simple flow chart:

```mermaid
sequenceDiagram
    participant browser
    participant server

    browser->>server:Executes the JavaScript code fetched from the server
    activate server
    server-->>browser:fetches the notes as JSON data and adds HTML elements for displaying the notes
    deactivate server

    Note left of server:uses DOM-API

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/notes
    activate server
    server-->>browser: the contents of are manipulated with JavaScript that executes in the browser.
    deactivate server

    Note right of browser:comprise only one HTML page

    browser->>server:data is sent via form's submit
    activate server
    server-->>browser:instructs to reload the Notes page with a redirect
    deactivate server

    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/spa
    activate server
    server-->>browser:here,the JavaScript file is spa.js
    deactivate server
    
    browser->>server: On creation of new note,only one request is being sent
    activate server
    server-->>browser:responds with status code 201 created
    deactivate server

    Note left of server:does not ask for a redirect
    Note right of browser:stays on same page,and it sends no further HTTP requests

    browser->>server:note list on the page is rerenderedd and the new note is being sent
    activate server 
    server-->>browser:the data type is determined with a Content-type header and then it is sent as JSON string
    deactivate server

    browser->>server:the user goes to the 'spa' version of the notes app
    activate server
    server-->>browser:Executed the excercise:Single page app diagram
    deactivate server

    Note right of browser:Excercise 0.5
```

### CODE REVIEW

**Comment by ArpitAgrawal001** [comment]
Sir should i do some changes in my spa sequence diagram or is it ready to merge?

File: Excercise 0.5.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    deactivate server
+
+    Note right of browser:comprise only one HTML page
+
+    browser->>server:data is sent via form's submit
+    activate server
+    server-->>browser:instructs to reload the Notes page with a redirect
+    deactivate server
+
+    browser->>server: GET https://studies.cs.helsinki.fi/exampleapp/spa
```
Comment (mavenik) [inline Excercise 0.5.md]: @ArpitAgrawal001 This is where the single page app flow actually begins. Everything before this line is the traditional flow.

File: Excercise 0.5.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    activate server
+    server-->>browser:here,the JavaScript file is spa.js
+    deactivate server
+    
+    browser->>server: On creation of new note,only one request is being sent
```
Comment (mavenik) [inline Excercise 0.5.md]: @ArpitAgrawal001 New note creation using SPA is part of the next exercise. Any reason why it is here?

**Review by mavenik — CHANGES_REQUESTED** [review]
@ArpitAgrawal001 Exercise 0.5 is about understanding how a single page application is loaded in browser. Your sequence diagram initially talks about the traditional method without Javascript, then only mentions one request to SPA and it is followed by new note creation flow.

It is expected to list a series of network activity that occurs when SPA is loaded in browser. Can you update your implementation please?

Also, I see some unwanted code such as a React app, etc and missing part-specific directories which should contain the exercise files. Can you fix that too?

**Comment by ArpitAgrawal001** [comment]
Here is a simple flow chart:

```mermaid
sequenceDiagram
    participant Browser
    participant Server
    participant SPA

    Browser->>Server: HTTP request for https://studies.cs.helsinki.fi/exampleapp/spa
    activate Server
    
    Server-->>Browser: SPA files (HTML, CSS, JS) are served
    deactivate Server
    
    Browser->>SPA: Browser receives SPA files
    activate SPA
    
    SPA-->>Browser: SPA initializes and renders
    deactivate SPA
    
    Browser->>Server: AJAX requests for initial data/resources
    activate Server
    
    Server->>SPA: Initial data/resources are served
    deactivate Server
    activate SPA
    
    SPA-->>Browser:Executed the excercise:Single page app diagram
    deactivate SPA

    Note right of Browser:Excercise 0.5

```

**Comment by ArpitAgrawal001** [comment]
Sir i have done some modifications, Please have a look .
And sir inn develop branch also i have done some modifications and created a directory for part0 and have excercise0.4 file in it,
On the other hand on my spa feature branch I have only kept my excercise_0.5.md file.

Sir is it ready to merge, or are there some more changes needed.
And sir after merging should i delete my both the feature branches (New_note_diagram and Spa diagram).

**Comment by ArpitAgrawal001** [comment]
Sir the commits which are being shown before my updated flowchart are not based on this, I have done those commits and pushes to correct some of my wrong commits.

Please do ignore it

**Comment by ArpitAgrawal001** [comment]
URL for the develop branch: 
https://github.com/ArpitAgrawal001/fullstackopen-submissions/tree/develop

URL for the spa branch:
https://github.com/ArpitAgrawal001/fullstackopen-submissions/tree/Single_page_app_diagram

**Review by mavenik — APPROVED** [review]
Looks okay to me. Can be merged.

**Pull request merged**